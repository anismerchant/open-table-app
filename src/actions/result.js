const getResult = ({ data }) => ({
  type: 'GET_RESTAURANT',
  data
});

export default getResult;