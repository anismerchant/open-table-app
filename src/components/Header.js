import React from 'react'
import { PageHeader } from 'antd'

const styles = {
  pageHeaderContainer: {
    border: '1px solid #D9D9D9',
    marginBottom: 10,
  }
}

export default () => (
  <PageHeader 
    style={styles.pageHeaderContainer} 
    title="OpenTable" 
    subTitle="Find restaurants near you" 
  />
)
