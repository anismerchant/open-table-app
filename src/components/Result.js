import React from 'react';
import { Card } from 'antd';

const { Meta } = Card;

export default (props) => (     
  <Card 
    hoverable
    style={{ width: 240, margin: 15 }}
    cover={<img alt="restaurant" src={props.image_url}  />}
  >
    <Meta title={props.name} description={props.address} />
    <p>{props.city}, {props.state} {props.postal_code}</p>
    <p>${props.price}</p>
  </Card>
);

