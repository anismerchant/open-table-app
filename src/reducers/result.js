const resultDefaultState = {}

const resultReducer = (state=resultDefaultState, action) => {
  switch(action.type) {
    case 'GET_RESTAURANT':
      return action.data
    default:
      return state
  }
}

export default resultReducer
