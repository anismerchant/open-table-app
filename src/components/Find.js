import React, { Component } from 'react'
import { connect } from 'react-redux'
import axios from 'axios'
import { BackTop, Empty, Input } from 'antd'
import getResult from '../actions/result'
import Result from './Result'

const { Search } = Input

const apiEndpoint = 'https://opentable.herokuapp.com/api/restaurants'
const queryString = '?city='

const styles = {
  cardContaier: {
    display: 'flex',
    flexWrap: 'wrap',
    justifyContent: 'center',
    alignItem: 'center',
  },
  resultContainer: {
    margin: 5
  },
  bodyContainer: {
    height: 'auto',
    position: 'relative', 
  },
  imgContainer: {
    margin: 'auto',
    width: '95%',
    maxHeight: '35rem',
    opacity: '0.7',
    filter: 'brightness(60%) contrast(90%)',
  },
  text: {
    fontSize: '4vh',
    color: 'white',
    fontWeight: 'bold',
    position: 'absolute',
    top: '25%',
    left: '8%',
    width: '80%',
    lineHeight: 1,
  },
  searchContainer: {
    width: '70%',
    height: 40,
    position: 'absolute',
    top: '60%',
    left: '16%',
  },
  resultHeading: {
    title: {
      display: 'inline-block',
      fontSize: '3vh',
    },
    city: {
      display: 'inline-block',
      paddingLeft: 10,
      fontWeight: 'bold',
      fontSize: '3vh',
    },
  }
}

export class Find extends Component {
  state = {
    city: ''
  }

  // Capture city input
  onChange = (e) => {
    const city = e.target.value
    if(!city) {
      this.setState(() => ({ city: '' }))
      return this.clearResult()
    }
    this.setState(() => ({ city }))
  }

  // Api call
  onKeyUp = () => {
    if (this.state.city) {
      axios.get(`${apiEndpoint}${queryString}${this.state.city}`)
      .then(response => 
        this.props.onKeyUp({data: response.data})
      )
      .catch(console.log)
    }
  }

  // Display results
  showResult = () => {
    const  {restaurants } = this.props.result
    if(!restaurants || !this.state.city) return
    if(!restaurants.length) {
      return <Empty image={Empty.PRESENTED_IMAGE_SIMPLE} description='No Restaurants Found.'/>
    }
    return restaurants.map(restaurant => {
      return <Result style={styles.resultContainer} key={restaurant.id} {...restaurant} />
    }) 
  }

  // Clear results
  clearResult = () => {
    if(!this.state.city) {
      return this.showResult()
    }
  }

  render()  {
    return (
      <div>   
        <form onKeyUp={this.onKeyUp} onSubmit={(e) => e.preventDefault()}>
          <div style={styles.bodyContainer}>
            <img style={styles.imgContainer} src='./assets/images/restaurant-large.jpg' alt="restaurant" />
            <p style={styles.text}>Find your table for any occasion</p>
            <Search
              type="text"
              placeholder="City"
              autoFocus
              onChange={this.onChange}
              style={styles.searchContainer}
            />
          </div>
        </form>
        <p style={styles.resultHeading.title}>Popular restaurants in</p>
        <div style={styles.resultHeading.city}>{this.state.city}</div>
        <div style={styles.cardContaier}>
          {this.showResult()}
        </div>
        <div>
          <BackTop />
        </div>
      </div>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    result: state.result
  }
}

const mapDispatchToProps = (dispatch) => ({
  onKeyUp: (payload) => dispatch(getResult(payload)) 
})

export default connect(mapStateToProps, mapDispatchToProps)(Find)
