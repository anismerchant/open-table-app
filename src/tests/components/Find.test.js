import React from 'react';
import { shallow } from 'enzyme';
import { Find } from '../../components/Find';
import result from '../fixtures/result';

test('It should correctly render Find.', () => {
  const wrapper = shallow(<Find result={result}  />);
  expect(wrapper).toMatchSnapshot();
});