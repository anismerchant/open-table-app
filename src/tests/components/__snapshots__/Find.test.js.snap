// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should correctly render Find. 1`] = `
<div>
  <form
    onKeyUp={[Function]}
    onSubmit={[Function]}
  >
    <div
      style={
        Object {
          "height": "auto",
          "position": "relative",
        }
      }
    >
      <img
        alt="restaurant"
        src="./assets/images/restaurant-large.jpg"
        style={
          Object {
            "filter": "brightness(60%) contrast(90%)",
            "margin": "auto",
            "maxHeight": "35rem",
            "opacity": "0.7",
            "width": "95%",
          }
        }
      />
      <p
        style={
          Object {
            "color": "white",
            "fontSize": "4vh",
            "fontWeight": "bold",
            "left": "8%",
            "lineHeight": 1,
            "position": "absolute",
            "top": "25%",
            "width": "80%",
          }
        }
      >
        Find your table for any occasion
      </p>
      <Search
        autoFocus={true}
        enterButton={false}
        onChange={[Function]}
        placeholder="City"
        style={
          Object {
            "height": 40,
            "left": "16%",
            "position": "absolute",
            "top": "60%",
            "width": "70%",
          }
        }
        type="text"
      />
    </div>
  </form>
  <p
    style={
      Object {
        "city": Object {
          "display": "inline-block",
          "fontSize": "3vh",
          "fontWeight": "bold",
          "paddingLeft": 10,
        },
        "display": "inline-block",
        "fontSize": "3vh",
      }
    }
  >
    Popular restaurants in
  </p>
  <div
    style={
      Object {
        "display": "inline-block",
        "fontSize": "3vh",
        "fontWeight": "bold",
        "paddingLeft": 10,
      }
    }
  />
  <div
    style={
      Object {
        "alignItem": "center",
        "display": "flex",
        "flexWrap": "wrap",
        "justifyContent": "center",
      }
    }
  >
    <div />
  </div>
</div>
`;
